sdk/rust/oak/src/grpc/mod.rs

```diff
@@ -162,6 +162,94 @@ pub trait OakNode {
     fn invoke(&mut self, method: &str, req: &[u8], writer: ChannelResponseWriter);
 }
 
+/// Encapsulate a protocol buffer message in a GrpcRequest wrapper using the
+/// given method name.
+pub fn encap_request<T: protobuf::Message>(req: T, method_name: &str) -> Option<Vec<u8>> {
+    // Put the request in a GrpcRequest wrapper and serialize it.
+    let mut grpc_req = GrpcRequest::new();
+    grpc_req.set_method_name(method_name.to_string());
+    let mut any = protobuf::well_known_types::Any::new();
+    if let Err(e) = req.write_to_writer(&mut any.value) {
+        warn!("failed to serialize gRPC request: {}", e);
+        return None;
+    };
+    grpc_req.set_req_msg(any);
+    grpc_req.set_last(true);
+    let mut buf = Vec::new();
+    grpc_req
+        .write_to_writer(&mut buf)
+        .expect("failed to serialize GrpcRequest message");
+    Some(buf)
+}
+
+/// Helper to inject a (single) gRPC request message via a notification channel,
+/// in the same manner as the gRPC pseudo-Node does, and collect a (single)
+/// response.
+pub fn invoke_grpc_method<R, Q>(
+    method_name: &str,
+    req: R,
+    invocation_channel: &crate::io::Sender<Invocation>,
+) -> Result<Q>
+where
+    R: protobuf::Message,
+    Q: protobuf::Message,
+{
+    // Put the request in a GrpcRequest wrapper and serialize into a message.
+    let req_data = encap_request(req, method_name).expect("failed to serialize GrpcRequest");
+
+    // Create a new channel to hold the request message.
+    let (req_write_half, req_read_half) =
+        crate::channel_create().expect("failed to create channel");
```

Comment:
At some point we should have a wrapper for `channel_create` that returns corresponding `Sender` and `Receiver` for a given type, I think it would also make it less verbose to implement my other suggestions here.

Reply:
Done as `oak::io::channel_create<T>`